test/app/post_stream_test.dart

```diff
@@ -0,0 +1,165 @@
+import 'package:black_tax_and_white_benefits/app/data/post_client.dart';
+import 'package:black_tax_and_white_benefits/app/domain/post.dart';
+import 'package:black_tax_and_white_benefits/app/view/app.dart';
+import 'package:black_tax_and_white_benefits/app/view/post_cell.dart';
+import 'package:flutter/material.dart';
+import 'package:flutter_test/flutter_test.dart';
+import 'package:hooks_riverpod/hooks_riverpod.dart';
+import 'package:mocktail/mocktail.dart';
+
+class MockPostClient extends Mock implements PostClient {}
+
+class Listener<T> extends Mock {
+  void call(T? previous, T next);
+}
+
+void main() {
+  ProviderContainer createContainer(PostClient postClient) {
+    final container = ProviderContainer(
+      overrides: [
+        getPostsProvider.overrideWith((ref) async {
+          await Future<void>.delayed(const Duration(seconds: 1));
+          return postClient.getPosts(100);
+        }),
+      ],
+    );
+    addTearDown(container.dispose);
+    return container;
+  }
+
+  Future<void> pumpApp(WidgetTester tester, PostClient postClient) async {
+    await tester.pumpWidget(
+      ProviderScope(
+        overrides: [
+          getPostsProvider.overrideWith((ref) async {
+            await Future<void>.delayed(const Duration(seconds: 1));
+            return postClient.getPosts(100);
+          }),
+        ],
+        child: const App(),
+      ),
+    );
+  }
+
+  setUpAll(() {
+    registerFallbackValue(const AsyncData<List<Post>>([]));
+  });
+
+  group('Test PostClient and getPosts Provider', () {
+    test('Test posts success', () async {
+      final postClient = MockPostClient();
+
+      final data = <Post>[
+        const Post(
```

Comment:
this is probably an example where yhou can reuse the @Jsonliteral and save lines of code and such

Reply:
done